pkg/clusteragent/autoscaling/workload/controller.go

```diff
@@ -272,8 +280,20 @@ func (c *Controller) syncPodAutoscaler(ctx context.Context, key, ns, name string
 	// Reaching this point, we had an error in processing, clearing up global error
 	podAutoscalerInternal.SetError(nil)
 
-	// Now that everything is synced, we can perform the actual processing
-	result, err := c.handleScaling(ctx, podAutoscaler, &podAutoscalerInternal)
+	isAdded := c.AutoscalerHeap.InsertIntoHeap(autoscaling.TimestampKey{
+		Timestamp: podAutoscalerInternal.CreationTimestamp(),
+		Key:       key,
+	})
+
+	result := autoscaling.NoRequeue
+	var err error
+	if !isAdded {
+		// Number of DatadogPodAutoscaler objects exceeds the limit
+		podAutoscalerInternal.SetError(fmt.Errorf("Too many DatadogPodAutoscaler objects created, ignoring this one %s", key))
```

Comment:
I think we need to be a bit more explicit. You also don't need to put the `key` in the error message as it's already attached to the object itself.
Something like `Autoscaler disabled as maximum number per cluster reached (100)`.